private/serialization.nim

```diff
@@ -62,12 +62,40 @@ proc representObject*(value: string, ts: TagStyle,
   ## represents a string as YAML scalar
   c.put(scalarEvent(value, tag, yAnchorNone))
 
+proc parseHex[T: int8|int16|int32|int64|uint8|uint16|uint32|uint64](s: string): T =
+  result = 0
+  var i = 2
+  while true:
```

Comment:
ah yes, just saw that it is because you start at two. you can use `for i in 2..<s.len` then.